server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
       @PathParam("dataSourceName") String dataSourceName,
-      @QueryParam("interval") String interval
+      @QueryParam("interval") String interval,
+      @QueryParam("full") String full
   )
   {
-    return ImmutableMap.<String, Object>of(
-        "dimensions", getDatasourceDimensions(dataSourceName, interval),
-        "metrics", getDatasourceMetrics(dataSourceName, interval)
-    );
+    if(full == null) {
+      return ImmutableMap.<String, Object>of(
+          KEY_DIMENSIONS, getDatasourceDimensions(dataSourceName, interval),
+          KEY_METRICS, getDatasourceMetrics(dataSourceName, interval)
+      );
+    }
+
+    Interval theInterval;
+    if (interval == null || interval.isEmpty()) {
+      DateTime now = new DateTime();
+      theInterval = new Interval(now.minusMillis(SEGMENT_HISTORY_MILLIS), now);
```

Comment:
i think the history can be a queryparam and we can have a default value if it is not included